agent/kvs_endpoint.go

```diff
@@ -46,7 +46,7 @@ func (s *HTTPHandlers) KVSEndpoint(resp http.ResponseWriter, req *http.Request)
 	if !s.agent.config.DisableHTTPUnprintableCharFilter && args.Key != "" {
 		// Allowed key pattern: a-zA-Z0-9 ,-_./
 		// https://developer.hashicorp.com/consul/docs/automate/kv#using-consul-kv
-		kvKeyPattern := `^[a-zA-Z0-9,_./\-?&=]+$`
+		kvKeyPattern := `^[a-zA-Z0-9,_./\-?&=+*%ç]+$`
```

Comment:
It seems kind of weird to include this `ç` character here when it is such a implementation specific thing, for that traefik key.

Is there not a better regex that we can construct that will include all possible URL valid characters, or maybe we should do the reverse, a regex that will match anything we explicitly shouldn't allow in a key value.

Reply:
I think if we're going to re-use the option value,[disable_http_unprintable_char_filter](https://developer.hashicorp.com/consul/docs/reference/agent/configuration-file/general#disable_http_unprintable_char_filter), then we should perform the same validation here as the other location that switches on this value.

https://github.com/hashicorp/consul/blob/main/agent/http.go#L325

Otherwise we should not use that option and this behavior should be behind a flag that is has the check disabled by default. That would have prevented many of us early adopters to not be presented w/ issues because of the poorly thought out addition.

Reply:
Personally I would like to re-enable disable_http_unprintable_char_filter and this PR does not make me want to do that.

Reply:
@sanikachavan5
@Vikramarjuna

Reply:
Please refer to the work done in https://github.com/hashicorp/consul/pull/22863 for more details.